client/src/modules/cash/payments/templates/search.modal.html

```diff
@@ -53,21 +53,11 @@
             <bh-clear on-clear="$ctrl.clear('user_id')"></bh-clear>
           </bh-user-select>
 
-          <div class="form-group">
-            <label class="control-label" translate>
-              FORM.LABELS.DEBTOR_GROUP
-            </label>
-            <bh-clear on-clear="$ctrl.clear('debtor_group_uuid')"></bh-clear>
-            <!-- @todo make data driven selects components to handle loading and error states -->
-            <ui-select
-              name="debtor_group_uuid"
-              ng-model="$ctrl.searchQueries.debtor_group_uuid">
-              <ui-select-match placeholder="{{ 'FORM.PLACEHOLDERS.DEBTOR_GROUP' | translate }}"><span>{{$select.selected.name}}</span></ui-select-match>
-              <ui-select-choices ui-select-focus-patch repeat="group.uuid as group in ($ctrl.debtorGroups | filter:$select.search | orderBy:'name') track by group.uuid">
-                <span ng-bind-html="group.name | highlight:$select.search"></span>
-              </ui-select-choices>
-            </ui-select>
-          </div>
+          <bh-debtor-group-select
+            debtor-group-uuid = "$ctrl.searchQueries.debtor_group_uuid"
+            on-select-callback = "$ctrl.onSelectDebtor(debtorGroup)">
+			<bh-clear on-clear="$ctrl.clear('debtor_group_uuid')"></bh-clear>
```

Comment:
This inner component (`bh-clear`) is not currently displayed. I think you have to allow the component (`bh-debtor-group-select`) to `transclude`